coral/src/app/features/topics/schema-request/components/TopicSchema.tsx

```diff
@@ -0,0 +1,151 @@
+        return await readFile(file);
+      }
+      getFile().then((fileContent) => {
+        clearErrors();
+        setSchema(fileContent);
+      });
+    } else {
+      setError(name, { message: "File is a required field" });
+      setSchema(undefined);
+    }
+  }
+
+  return (
+    <_Controller
+      name={name}
+      control={control}
+      render={({ fieldState: { error } }) => {
+        return (
+          <div>
+            <FileInput
+              {...props}
+              buttonText={"Upload AVRO schema"}
+              labelText={"Upload AVRO schema file"}
+              noFileText={"No file chosen"}
+              helperText={error?.message || ""}
+              required={required}
+              onBlur={checkEmptyFile}
+              onChange={uploadFile}
+              valid={!error}
+            />
+            <Typography.Caption fontWeight={"500"} htmlTag={"label"}>
+              <span className={error?.message && "text-error-50"}>
+                Schema preview (read only)
+              </span>
+            </Typography.Caption>
+            <BorderBox
+              borderColor={error?.message ? "error-50" : "grey-20"}
+              marginTop={"1"}
+            >
+              {!schema && (
+                <div style={{ height: "200px" }}>
+                  <Box
+                    display={"flex"}
+                    justifyContent={"center"}
+                    alignItems={"center"}
+                    height={"full"}
+                    backgroundColor={"grey-5"}
+                    aria-hidden={"true"}
+                  >
+                    Preview for your schema
+                  </Box>
+                </div>
+              )}
+              <div style={{ display: `${!schema ? "none" : "block"}` }}>
+                <MonacoEditor
+                  data-testid="topic-schema"
+                  height="200px"
+                  language="json"
+                  theme={"light"}
+                  value={schema}
+                  onValidate={validateSchema}
+                  loading={"Loading preview"}
+                  options={{
+                    ariaLabel: "Preview",
+                    readOnly: true,
+                    domReadOnly: true,
+                    renderControlCharacters: false,
+                    minimap: { enabled: false },
+                    folding: false,
+                    lineNumbers: "off",
+                    scrollBeyondLastLine: false,
+                    scrollbar: {
+                      vertical: "hidden",
+                    },
+                  }}
+                />
+              </div>
+            </BorderBox>
+            <Box
+              component={"p"}
+              marginTop={"1"}
+              marginBottom={"3"}
+              className={"text-error-50 typography-caption-default"}
+              aria-hidden={"true"}
+            >
+              {error?.message ? `${error?.message}` : <>&nbsp;</>}
```

Comment:
Could we have conditional rendering for this whole `<Box` based on the existence of the error.

Reply:
The `Box` must be rendered with an `&nbsp;` so it creates a div with the same height as the text will take. Otherwise, there's a jump in the layout when an error is rendered. It's the exact implementation the DS uses!